tensorboard/webapp/metrics/effects/metrics_effects_test.ts

```diff
@@ -777,5 +782,108 @@ describe('metrics effects', () => {
         });
       }
     });
+
+    describe('loadSavedPins', () => {
+      let saveScalarPinSpy: jasmine.Spy;
+      let removeScalarPinSpy: jasmine.Spy;
+
+      beforeEach(() => {
+        saveScalarPinSpy = spyOn(savedPinsDataSource, 'saveScalarPin');
+        removeScalarPinSpy = spyOn(savedPinsDataSource, 'removeScalarPin');
+
+        store.overrideSelector(TEST_ONLY.getCardFetchInfo, {
+          id: 'card1',
+          plugin: PluginType.SCALARS,
+          tag: 'tagA',
+          runId: null,
+          loadState: DataLoadState.LOADED,
+        });
+        store.overrideSelector(
+          selectors.getVisibleCardIdSet,
+          new Set(['card1'])
+        );
+        store.refreshState();
+      });
+
+      it('removes scalar pin if the given card was pinned', () => {
+        actions$.next(
+          actions.cardPinStateToggled({
+            cardId: 'card1',
+            wasPinned: true,
+            canCreateNewPins: true,
+          })
+        );
+
+        expect(removeScalarPinSpy).toHaveBeenCalled();
```

Comment:
toHaveBeenCalled() is made redundant by the toHaveBeenCalledWith() below (same for other tests)

Reply:
Applied!